app/lib/puppet_fact_parser_extensions.rb

```diff
@@ -2,42 +2,28 @@ module PuppetFactParserExtensions
   extend ActiveSupport::Concern
 
   included do
-    # In Foreman 1.8 these two methods have been removed, we could reuse discovery_mac_fact_name in suggested_primary_interface(host) if needed
-    if instance_methods.include?(:primary_interface)
-      alias_method_chain :primary_interface, :discovery_fact
-    end
-
-    if instance_methods.include?(:ip)
-      alias_method_chain :ip, :discovery_fact
+    begin
+      raise "method: suggested_primary_interface" unless instance_methods.include?(:suggested_primary_interface)
+      raise "method: parse_interfaces?" unless instance_methods.include?(:parse_interfaces?)
```

Comment:
Is this required? Presumably, we're enforcing a version of Foreman in the engine.rb that has these methods.